kshape/core.py

```diff
@@ -155,34 +157,83 @@ def _extract_shape(idx, x, j, cur_center):
     return zscore(centroid, ddof=1)
 
 
-def _kshape(x, k):
+def _kshape(x, k,n_init=1, max_iter=100, n_jobs = 1, random_state=None):
     """
     >>> from numpy.random import seed; seed(0)
     >>> _kshape(np.array([[1,2,3,4], [0,1,2,3], [-1,1,-1,1], [1,2,2,3]]), 2)
     (array([0, 0, 1, 0]), array([[-1.2244258 , -0.35015476,  0.52411628,  1.05046429],
            [-0.8660254 ,  0.8660254 , -0.8660254 ,  0.8660254 ]]))
     """
-    m = x.shape[0]
-    idx = randint(0, k, size=m)
+    random_state = check_random_state(random_state)
+    best_tot_dist,best_centroids,best_idx = None,None,None
+    
+    if n_jobs ==1:
+
+        for i_init in range(n_init):    
+            # n_init is the number of random starting points
+            # pdb.set_trace()
+            
+            idx, centroids,tot_dist, iterations = _kshape_single(x, k, max_iter=max_iter, random_state= random_state) 
+            if best_tot_dist is None or tot_dist < best_tot_dist:
+                best_idx = idx.copy()
+                best_centroids = centroids.copy()
+                best_tot_dist = tot_dist
+    else: # n_jobs not =1 # if -1, all CPUs are used
```

Comment:
The semantics of `n_jobs` should be not hidden in comments, but publicly documented.